qiskit_nature/second_q/mappers/bosonic_linear_mapper.py

```diff
@@ -0,0 +1,165 @@
+import operator
+import warnings
+
+from functools import reduce, lru_cache
+
+import numpy as np
+
+from qiskit.quantum_info import Pauli, SparsePauliOp
+from qiskit_nature import settings
+
+from qiskit_nature.second_q.operators import BosonicOp
+from .bosonic_mapper import BosonicMapper
+
+
+class BosonicLinearMapper(BosonicMapper):
+    """
+   The Linear boson-to-qubit mapping.
+
+   This mapper generates a linear encoding of the Bosonic operator :math:`b_k^\\dagger, b_k` to qubit
+   operators (linear combinations of pauli strings).
+   In this linear encoding each bosonic mode is represented via :math:`n_k^{max} + 1` qubits, where
+   :math:`n_k^{max}` is the truncation of the mode (meaning the number of states used in the expansion of
+   the mode, or equivalently the state at which the maximum excitation can take place).
+   The mode :math:`|k\rangle` is then mapped to the occupation number vector
+   :math:`|0_{n_k^{max}}, 0_{n_k^{max} - 1}, ..., 0_{n_k + 1}, 1_{n_k}, 0_{n_k - 1}, ..., 0_{0_k}\rangle`
+
+   It implements the formula in Section II.C of Reference [1]:
+
+   .. math::
+       b_k^\\dagger = \\sum(\\sqrt{n_k + 1} \\sigma_{n_k}^+\\sigma_{n_k + 1}^-)
+
+   from :math:`n_k = 0` to :math:`n_k^max + 1`
+   where :math:`n_k^max` is the truncation order (defined by the user).
+   In the following implementation, we explicit the operators :math:`\\sigma^+` and :math:`\\sigma^-`
+   with the Pauli matrices:
+
+   .. math::
+       \\sigma_{n_k}^+ := S_j^+ = 0.5 * (X_j + i*Y_j)\\
+       \\sigma_{n_k}^- := S_j^- = 0.5 * (X_j - i*Y_j)
+
+   The length of the qubit register is: ``BosonicOp.num_modes * (BosonicLinearMapper.truncation + 1)``
+
+   References:
+       [1] A. Miessen et al., Quantum algorithms for quantum dynamics: A performance study on the
+       spin-boson model, Phys. Rev. Research 3, 043212.
+       https://link.aps.org/doi/10.1103/PhysRevResearch.3.043212
+
+    """
+
+    def _map_single(
+        self, second_q_op: BosonicOp, *, register_length: int | None = None
+    ) -> SparsePauliOp:
+        """Maps a :class:`~qiskit_nature.second_q.operators.SparseLabelOp` to a``SparsePauliOp``.
+
+        Args:
+            second_q_op: the ``SparseLabelOp`` to be mapped.
+            register_length: when provided, this will be used to overwrite the ``register_length``
+                attribute of the operator being mapped. This is possible because the
+                ``register_length`` is considered a lower bound in a ``SparseLabelOp``.
+
+        Returns:
+            The qubit operator corresponding to the problem-Hamiltonian in the qubit space.
+        """
+        if settings.use_pauli_sum_op:
+            warnings.warn(
+                DeprecationWarning(
+                    "As of version 0.6.0 the PauliSumOp is deprecated. We are ignoring this setting"
+                    "for this function. Thus the returned object will have type SparsePauliOp"
+                )
+            )
```

Comment:
I am not sure which error/warning we should raise here, but it is not necessarily a deprecation one, we are just warning that the setting should not be there. Sth like this (you will have to remove `import warnings` from the top or lint will complain):

```suggestion
            raise NotImplementedError("Setting not supported for this class. Please disable settings.use_pauli_sum_op.")
```

Reply:
Solved